presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSource.java

```diff
@@ -215,7 +216,7 @@ private static Object deserializePartitionValue(Type type, String valueString, S
                 return parseLong(valueString);
             }
             if (type.equals(TIMESTAMP)) {
-                return parseLong(valueString);
+                return MICROSECONDS.toMillis(parseLong(valueString));
```

Comment:
We should consider adding support for variable-precision timestamps (or, at least, expose this as `timestamp(6)`, since we're dealing with microseconds)

Reply:
I'll consult with @electrum as to how to get this done.

Reply:
Agreed. This should should simplify the various `TupleDomain` and value conversions, but will require support in the readers and writers for ORC and Parquet, so it's not a trivial project.